actionpack/lib/action_dispatch/system_test_case.rb

```diff
@@ -105,9 +105,12 @@ def self.start_application # :nodoc:
     #
     #   driven_by :selenium, screen_size: [800, 800]
     def self.driven_by(driver, using: :chrome, screen_size: [1400, 1400])
-      SystemTesting::Driver.new(driver).run
+      driver = selenium?(driver) ? SystemTesting::Browser.new(using, screen_size) : SystemTesting::Driver.new(driver)
+
+      setup { driver.use }
+      teardown { driver.reset }
```

Comment:
Capybara uses this pattern with RSpec - https://github.com/teamcapybara/capybara/blob/41542d5617f638608cff301b5b454b011ae41f53/lib/capybara/rspec.rb#L19-L31 - so maybe makes sense to follow it here. Not resetting the global setting might open space for misconfigured test cases if for instance someone writes a test case that inherits directly from `SystemTestCase` (instead of the `ApplicationSystemTestCase`) and forgets to add their own `driven_by` call to it.